srv/_modules/sspl.py

```diff
@@ -44,17 +44,27 @@ def merge_health_map_schema(source_json="/tmp/resource_health_view.json"):
   health_map_file = healthmap_schema_pillar['filename']
 
   data = local.cmd('*', 'file.read', [source_json])
-  node1_data = json.loads(data["srvnode-1"])
-  node2_data = json.loads(data["srvnode-2"])
+  node_data = []
+  for node in __pillar__['cluster'].keys():
+    if "srvnode-" in node:
+      node_data = json.loads(data[node])
+
+  node_rack_nodes = None
+  for i, node_data in enumerate(node_rack_nodes):
```

Comment:
I think this code will not work and will raise an Exception. I can guess that code supposed to be:
```python
for i, _data in enumerate(node_data):
    if i == 0:
      node_rack_nodes = (
        _data["cluster"]["sites"]["001"]["rack"]["001"]["nodes"]
      )
    else:
      node_rack_nodes.update(
        _data["cluster"]["sites"]["001"]["rack"]["001"]["nodes"]
      )
```

Reply:
Also it can be simplified like this
```python
node_rack_nodes = node_data[0]["cluster"]["sites"]["001"]["rack"]["001"]["nodes"]
for data in node_data:
   node_rack_nodes.update(data[i]["cluster"]["sites"]["001"]["rack"]["001"]["nodes"])
````

Reply:
`node_rack_nodes = node_data[0]["cluster"]["sites"]["001"]["rack"]["001"]["nodes"]`
Yes, initially thought about the same, but a null  value for _data would fail. A for loop in a way covers for the null value check.

Reply:
> I think this code will not work and will raise an Exception. I can guess that code supposed to be:
> 
> ```python
> for i, _data in enumerate(node_data):
>     if i == 0:
>       node_rack_nodes = (
>         _data["cluster"]["sites"]["001"]["rack"]["001"]["nodes"]
>       )
>     else:
>       node_rack_nodes.update(
>         _data["cluster"]["sites"]["001"]["rack"]["001"]["nodes"]
>       )
> ```

This was a buggy code and would have never worked. Thanks for pointing it out.
I've corrected it.